orte/mca/rml/ofi/rml_ofi_component.c

```diff
@@ -804,90 +868,99 @@ rml_ofi_component_init(int* priority)
             * Enable the endpoint for communication
             * This commits the bind operations.
             */
-            ret = fi_enable(orte_rml_ofi.ofi_conduits[cur_conduit].ep);
+            ret = fi_enable(orte_rml_ofi.ofi_prov[cur_ofi_prov].ep);
             if (0 != ret) {
                 opal_output_verbose(1, orte_rml_base_framework.framework_output,
                                     "%s:%d: fi_enable failed: %s\n",
                                     __FILE__, __LINE__, fi_strerror(-ret));
-                free_conduit_resources(cur_conduit);
+                free_ofi_prov_resources(cur_ofi_prov);
                 /* abort this current transport, but check if next transport can be opened */
                 continue;
             }
             opal_output_verbose(10,orte_rml_base_framework.framework_output,
-                            "%s:%d ep enabled for conduit_id - %d ",__FILE__,__LINE__,orte_rml_ofi.ofi_conduits[cur_conduit].conduit_id);
+                            "%s:%d ep enabled for ofi_prov_id - %d ",__FILE__,__LINE__,orte_rml_ofi.ofi_prov[cur_ofi_prov].ofi_prov_id);
 
 
             /**
             * Get our address and publish it with modex.
             **/
-            orte_rml_ofi.ofi_conduits[cur_conduit].epnamelen = sizeof (orte_rml_ofi.ofi_conduits[cur_conduit].ep_name);
-            ret = fi_getname((fid_t)orte_rml_ofi.ofi_conduits[cur_conduit].ep,
-                            &orte_rml_ofi.ofi_conduits[cur_conduit].ep_name[0],
-                            &orte_rml_ofi.ofi_conduits[cur_conduit].epnamelen);
+            orte_rml_ofi.ofi_prov[cur_ofi_prov].epnamelen = sizeof (orte_rml_ofi.ofi_prov[cur_ofi_prov].ep_name);
+            ret = fi_getname((fid_t)orte_rml_ofi.ofi_prov[cur_ofi_prov].ep,
+                            &orte_rml_ofi.ofi_prov[cur_ofi_prov].ep_name[0],
+                            &orte_rml_ofi.ofi_prov[cur_ofi_prov].epnamelen);
             if (ret) {
                         opal_output_verbose(1, orte_rml_base_framework.framework_output,
                                     "%s:%d: fi_getname failed: %s\n",
                                     __FILE__, __LINE__, fi_strerror(-ret));
-                free_conduit_resources(cur_conduit);
+                free_ofi_prov_resources(cur_ofi_prov);
                 /* abort this current transport, but check if next transport can be opened */
                   continue;
             }
 
-            switch ( orte_rml_ofi.ofi_conduits[cur_conduit].fabric_info->addr_format)
-            {
-                case  FI_SOCKADDR_IN :
-                    opal_output_verbose(1,orte_rml_base_framework.framework_output,
+            /* Register the ofi address of this peer with PMIX server only if it is a user process  /
+             * for daemons the set/get_contact_info is used to exchange this information */
+            if (ORTE_PROC_IS_APP) {
+                switch ( orte_rml_ofi.ofi_prov[cur_ofi_prov].fabric_info->addr_format)
```

Comment:
Should you use `fi_av_straddr()` here so that you don't have to know every type of libfabric address format?